packages/edit-site/src/components/post-list/index.js

```diff
@@ -353,6 +353,36 @@ export default function PostList( { postType } ) {
 		[ postTypeActions, editAction ]
 	);
 
+	const additionalContext = useSelect( ( select ) => {
+		const site = select( coreStore ).getEntityRecord( 'root', 'site' );
+		const themeTemplates = select( coreStore ).getEntityRecords(
+			'postType',
+			'wp_template',
+			{
+				per_page: -1,
+			}
+		);
+		const hasFrontPageTemplate = !! themeTemplates?.find(
+			( template ) => 'slug' in template && template.slug === 'front-page'
```

Comment:
Why not do a server side query instead of doing an unbound request?

Reply:
This was copied from `resolveTemplateForPostTypeAndId` [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js#L104). 

A server side query sounds good. Could you show me what you mean, or is there an example of a similar query somewhere?

Reply:
I've mentioned at https://github.com/WordPress/gutenberg/pull/65426/files#r1841865571 but this check can live in `packages/editor/src/dataviews/store/private-actions.ts` so the action is not even registered if the theme has a front page template (as opposed to checking the same thing for every item).